api/component_users.py

```diff
@@ -48,15 +50,17 @@ def do_get(self, **kwargs):
   @permissions.require_admin_site
   def do_put(self, **kwargs) -> tuple[dict, int]:
     params = self.request.get_json(force=True)
+    component_users_request = ComponentUsersRequest(**params)
```

Comment:
Let's try `ComponentUsersRequest.from_dict(params)` because that method does validation of the type of the user input.